lib/smb2/packet.rb

```diff
@@ -47,17 +47,134 @@ class InvalidFlagError < StandardError; end
   # Constants
   ##
 
-  # Values for {QueryInfoRequest#info_type}
-  # @see https://msdn.microsoft.com/en-us/library/cc246557.aspx
-  QUERY_INFO_TYPES = {
-    # SMB2_0_INFO_FILE
-    FILE: 0x01,
-    # SMB2_0_INFO_FILESYSTEM
-    FILESYSTEM: 0x02,
-    # SMB2_0_INFO_SECURITY
-    SECURITY: 0x03,
-    # SMB2_0_INFO_QUOTA
-    QUOTA: 0x04
+  # Used in {CreateRequest#disposition}
+  CREATE_DISPOSITIONS = {
+    # If the file already exists, supersede it. Otherwise, create the file.
+    # This value SHOULD NOT be used for a printer object.<30>
+    FILE_SUPERSEDE: 0x00000000,
+    # If the file already exists, return success; otherwise, fail the
+    # operation. MUST NOT be used for a printer object.
+    FILE_OPEN: 0x00000001,
+    # If the file already exists, fail the operation; otherwise, create the
+    # file.
+    FILE_CREATE: 0x00000002,
+    # Open the file if it already exists; otherwise, create the file. This
+    # value SHOULD NOT be used for a printer object.<31>
+    FILE_OPEN_IF: 0x00000003,
+    # Overwrite the file if it already exists; otherwise, fail the operation. MUST
+    # NOT be used for a printer object.
+    FILE_OVERWRITE: 0x00000004,
+    # Overwrite the file if it already exists; otherwise, create the file.
+    # This value SHOULD NOT be used for a printer object.<32>
+    FILE_OVERWRITE_IF: 0x00000005,
+  }.freeze
+
+  # Used in {CreateRequest#create_options}
+  #
+  # https://msdn.microsoft.com/en-us/library/cc246502.aspx
+  CREATE_OPTIONS = {
+    FILE_DIRECTORY_FILE: 0x0000_0001,
```

Comment:
I don't think so. They are used as a bitmask, but the value itself is unimportant, as are the shift values. In fact, I think it's easier to verify correctness with the documentation as is.